youtube_dl/__init__.py

```diff
@@ -141,7 +147,22 @@ def _real_main(argv=None):
             parser.error('auto number start must be positive or 0')
     if opts.usetitle and opts.useid:
         parser.error('using title conflicts with using video ID')
-    if opts.username is not None and opts.password is None:
+    if opts.username is not None and opts.password is None and opts.password_from_keyring is True:
+        # extract domain names, check if all videos are from the same domain.
+        all_domains = set(map(lambda url: compat_urllib_parse.urlparse(url).netloc, all_urls))
```

Comment:
No. Extractors are identified by special keys not domains, see `_NETRC_MACHINE`.